Update teamcity docs with info about fetchallheads (#1840)
GitTools/GitVersion: docs/build-server-support/build-server/teamcity.md

 - [Project Link](https://github.com/JetBrains/meta-runner-power-pack/tree/master/gitversion)

## Running inside TeamCity
When running in TeamCity you have two options, run using **agent checkout** or use dynamic repositories.

### Agent checkout
For GitVersion to pick up pull requests properly you need to promote the `%teamcity.build.vcs.branch.{configurationid}%` variable to an environment variable called `Git_Branch`

Just go to your build configuration, Parameters, click Add, Name should be `env.Git_Branch`, value should be `%teamcity.build.vcs.branch.{vcsid}%` where vcsid is your VCS root id. You should get auto completion for this.

> For GitVersion to work with [MainLine mode](../../reference/mainline-development.md) you need to set configuration parameter `teamcity.git.fetchAllHeads = true`, because TeamCity by default fetches only current branch for building, but MainLine mode requires master for calculating version.

Reviewer: ```suggestion
For GitVersion to work with any mode requiring other than the currently built branch to calculate the version number, you need to set the configuration parameter [`teamcity.git.fetchAllHeads = true` in TeamCity](https://www.jetbrains.com/help/teamcity/git.html#Git-GeneralSettings), because TeamCity by default fetches only the current branch for building.
```

> More reference about this #912, #1199, #1234, #1441, #1477 and #1825

Reviewer: These issue references won't resolve on [ReadTheDocs](https://gitversion.readthedocs.io/), so I think it's better to just reference them in the description of this PR rather than in the documentation itself.

Author: References removed and added to PR description

> Again go to your build configuration, Parameters, click Add, Name should be `teamcity.git.fetchAllHeads`, value should be `true`.

Reviewer: ```suggestion
To add this configuration parameter to your build configuration, go to *Parameters*, click *Add*, *Name* should be `teamcity.git.fetchAllHeads` and the value should be `true`.
```
